clustermesh/certs.go

```diff
@@ -183,21 +183,23 @@ func (k *K8sClusterMesh) deleteCertificates(ctx context.Context) error {
 }
 
 func (k *K8sClusterMesh) installCertificates(ctx context.Context) error {
-	err := k.certManager.LoadCAFromK8s(ctx)
-	if err != nil {
-		if !k.params.CreateCA {
-			k.Log("❌ Cilium CA not found: %s", err)
-			return err
-		}
 
+	caSecret, err := k.certManager.GetCASecret(ctx)
```

Comment:
Sounds good to me. I had initially intended to have all of that under the `GetCASecret` function but couldn't use the logger to tell the user we were generating the certificate in CertManager. I've changed the logging message after `LoadCAFromK8s` to accommodate both of these functions (since success of both of those functions are really all a user cares about) and moved the cert generation code into `GetCASecret`.